src/main/java/scenes/DemoTopDown.java

```diff
@@ -8,69 +8,74 @@
-
-		player = new GameObject("Player", new Transform(600, 600, 100, 100), 2);
-		player.addComponent(new PointLight(new Color(250, 255, 181), 30));
-		player.addComponent(new AABB());
-		player.addComponent(new SpriteRenderer(a.getSprite(132)));
-		player.addComponent(new CharacterController());
-
-		greenLight = new GameObject("Green light", new Transform(3315, 300, 1, 1), 3);
-		greenLight.addComponent(new PointLight(new Color(102, 255, 102), 30));
-
-		bloom = new BloomEffect(PostProcessStep.Target.DEFAULT_FRAMEBUFFER);
-		bloom.init();
-	}
-
-	public void update() {
-		super.update();
-		player.getComponent(PointLight.class).intensity = Utils.map((float) Math.sin(Engine.millis() / 600), -1, 1, 100, 140);
-		booper.getComponent(PointLight.class).intensity = Utils.map((float) Math.cos(Engine.millis() / 600), -1, 1, 70, 110);
-		greenLight.getComponent(PointLight.class).intensity = Utils.map((float) Math.cos(Engine.millis() / 600), -1, 1, 70, 110);
-
-		camera.smoothFollow(player.getTransform());
-	}
-
-	@Override
-	public void postProcess(int texture) {
-		bloom.apply(texture);
-	}
+    public static void main(String[] args) {
+        Engine.init("Azurite Engine Demo 1", 0.01f);
+        Engine.scenes().switchScene(new DemoTopDown(), true);
+        Engine.showWindow();
+    }
+
+    Spritesheet a;
+    Spritesheet b;
+    Tilesystem t;
+    GameObject player;
+    GameObject booper;
+    GameObject greenLight;
+    GameObject trRes;
+
+    BloomEffect bloom;
+
+    public void awake() {
+        camera = new Camera();
+        setDefaultBackground(0);
+
+        a = new Spritesheet(Assets.getTexture("src/assets/images/tileset.png"), 16, 16, 256, 0);
+        b = new Spritesheet(Assets.getTexture("src/assets/images/walls.png"), 16, 16, 256, 0);
+        t = new Tilesystem(this, a, b, 31, 15, 200, 200);
+
+        trRes = new GameObject("", new Transform(new Vector2f(0, 0), new Vector2f(100)), -20);
+
+        booper = new GameObject("Booper", new Transform(800, 800, 100, 100), 2);
+        booper.addComponent(new Animation(1, a.getSprite(132), a.getSprite(150)));
+        booper.addComponent(new CollisionTrigger(data -> System.out.println("Boop")));
+        booper.addComponent(new PointLight(new Color(255, 153, 102), 30));
+
+        player = new GameObject("Player", new Transform(600, 600, 100, 100), 2);
+        player.addComponent(new PointLight(new Color(250, 255, 181), 30));
+        player.addComponent(new AABB());
+        player.addComponent(new SpriteRenderer(a.getSprite(132)));
+        player.addComponent(new CharacterController());
+
+        greenLight = new GameObject("Green light", new Transform(3315, 300, 1, 1), 3);
+        greenLight.addComponent(new PointLight(new Color(102, 255, 102), 30));
+
+        bloom = new BloomEffect(PostProcessStep.Target.DEFAULT_FRAMEBUFFER);
+        bloom.init();
+
+        addGameObjectsToScene(trRes, booper, player, greenLight);

```

Comment:
if possible, I would like to move this method back to the GameObject constructor like it was before.
This is a somewhat unnecessary step that doesn't need to be exposed to the end user/developer.

Reply:
Fine, but the method to add an Object will still be visible and usuable by the user, cause GameObject and Scene are not in the same package so that making it protected isnt possible.

Reply:
that is alright, we can improve that later on.